src/main/java/commands/Command.java

```diff
@@ -0,0 +1,273 @@
+            currTask.setUndone();
+            return "    " + currTask;
+        } catch (IndexOutOfBoundsException var6) {
+            String single = tasks.size() <= 1 ? "task" : "tasks";
+            int num = tasks.size();
+            return "You only have " + num + " "
+                    + single
+                    + " currently. Type \"list\" to view all your current "
+                    + single;
+        }
+    }
+
+    public void delete(TaskList tasks, String fullCommand) {
+        String[] command = fullCommand.split(" ");
+        String single = tasks.size() <= 1 ? "task" : "tasks";
+        try {
+            Task currTask = tasks.getTask(Integer.parseInt(command[1]) - 1);
+            System.out.println("Noted. I've removed this task:");
+            tasks.deleteTask(Integer.parseInt(command[1]) - 1);
+            System.out.println("    " + currTask.toString());
+        } catch (IndexOutOfBoundsException var6) {
+            System.out.println("You only have " + tasks.size() + " " + single + " currently. Type \"tasks\" to view all your current " + single);
+        }
+        System.out.println("Now you have " + tasks.size() + " " + single + " in the tasks.");
+    }
+
+    public boolean bye() {
+        return false;
+    }
+
+    public void execute(String fullCommand, TaskList tasks, Ui ui, Storage store) {
+        String[] command = this.getKeywords(fullCommand);
+        if (command.length <= 1) {
+            ui.listOfCommands();
+        } else {
+            switch (command[0]) {
+            case "list":
+                ui.output(tasks.getList().toString());
+                break;
+            case "todo":
+            case "event":
+            case "deadline":
+                ui.showLine();
+                System.out.println("Got it. I've added this task: ");
+                String symbol = this.getTaskSymbol(command[0]);
+                int byStartIndex = fullCommand.indexOf("/by");
+                int fromStartIndex = fullCommand.indexOf("/from");
+                if (byStartIndex == -1 && fromStartIndex == -1) {
+                    String des = fullCommand.substring(command[0].length() + 1);
+                    Task newTask = new Task(des, symbol, false);
+                    tasks.addTask(newTask);
+                } else if (fromStartIndex == -1) {
+                    byCommand(fullCommand, byStartIndex, symbol, tasks);
+                } else {
+                    fromCommand(fullCommand, fromStartIndex, symbol, tasks);
+                }
+
+                Task latestTask = tasks.getTask(tasks.size() - 1);
+                String task = latestTask.toString();
+                System.out.println("    " + task);
+                String singular = tasks.size() == 1 ? "task" : "tasks";
+                int num = tasks.size();
+                System.out.println("Now you have " + num + " " + singular + " in the list.");
+                ui.showLine();
+                break;
```

Comment:
Maybe can add helper functions to handle this part so that your execute function can be shorter?

Reply:
Thanks for pointing it out, I have abstracted some of the logic away into a helper function.